app/containers/SignInPage/index.js

```diff
@@ -0,0 +1,89 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { compose } from 'redux';
+import { createStructuredSelector } from 'reselect';
+
+import injectReducer from '../../utils/injectReducer';
+import reducer from './reducer';
+import injectSaga from '../../utils/injectSaga';
+import saga from './saga';
+import { requestSignIn } from './actions';
+import {
+  makeSelectToken,
+  makeSelectMessage,
+} from './selectors';
+import SignInForm from '../../components/SignInForm/SignInForm';
+
+class SignInPage extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      email: '',
+      password: '',
+    };
+  }
+
+  changeInput(input, value) {
+    this.setState({ [input]: value });
+  }
+
+  handleButtonClick() {
+    const { submitForm } = this.props;
+    const { email, password } = this.state;
+
+    submitForm({ email, password });
+  }
+
+  render() {
+    return (
+      <SignInForm
+      {...{
+        onEmailChange: e => this.changeInput('email', e.target.value),
```

Comment:
Don't pass the event so far up in the component tree. Call `onEmailChange` directly with the new value.